lib/teleterm/clusters/cluster_headless.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusters
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/client"
+)
+
+// WatchPendingHeadlessAuthentications watches the backend for pending headless authentication requests for the user.
+func (c *Cluster) WatchPendingHeadlessAuthentications(ctx context.Context) (types.Watcher, error) {
+	var proxyClient *client.ProxyClient
+
+	err := addMetadataToRetryableError(ctx, func() error {
+		var err error
+		proxyClient, err = c.clusterClient.ConnectToProxy(ctx)
```

Comment:
Unlike in other places, we don't close `proxyClient` and `rootClient` here. I assume this is because the watcher is long-lived? How do we take care of cleaning up those clients?